src/bm_sim/match_tables.cpp

```diff
@@ -51,7 +51,7 @@ create_match_unit(const std::string match_type, const size_t size,
   else if (match_type == "lpm")
     match_unit = std::unique_ptr<MULPM>(
         new MULPM(size, match_key_builder, lookup_factory));
-  else if (match_type == "ternary")
+  else if ((match_type == "ternary") || (match_type == "optional"))
```

Comment:
IIRC this is not a correct change. The "match_type" here is the match type of the table, not the match type of an individual field. And a table is never going to have the optional match type, it will just be ternary.

Reply:
By the way, this match type is determined by the compiler backend.